tests/unit/test_exec.py

```diff
@@ -1,9 +1,50 @@
 __copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from jina.executors import BaseExecutor
+import pytest
+from jina import Document, DocumentArray
+from jinahub.encoder.laser_encoder import LaserEncoder
 
 
-def test_exec():
-    ex = BaseExecutor.load_config('../../config.yml')
-    assert ex._dim == 128
+@pytest.fixture()
+def docs_generator():
+    return DocumentArray((Document(text='random text') for _ in range(30)))
+
+
+def test_flair_batch(docs_generator):
+    encoder = LaserEncoder()
+    docs = docs_generator
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['r']})
+
+    assert len(docs.get_attributes('embedding')) == 30
+    assert docs[0].embedding.shape == (1024,)
+
+
+def test_traversal_path():
+    text = 'blah'
+    docs = DocumentArray([Document(id='root1', text=text)])
+    docs[0].chunks = [Document(id='chunk11', text=text),
+                      Document(id='chunk12', text=text),
+                      Document(id='chunk13', text=text)
+                      ]
+    docs[0].chunks[0].chunks = [
+        Document(id='chunk111', text=text),
+        Document(id='chunk112', text=text),
+    ]
+
+    encoder = LaserEncoder()
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['c']})
+
+    for path, count in [['r', 0], ['c', 3], ['cc', 0]]:
+        assert len(docs.traverse_flat([path]).get_attributes('embedding')) == count
+
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['cc']})
+    for path, count in [['r', 0], ['c', 3], ['cc', 2]]:
+        assert len(docs.traverse_flat([path]).get_attributes('embedding')) == count
```

Comment:
maybe add an assert on the shape of embeddings

Reply:
added